tests/e2e_test/module_deletion_test.go

```diff
@@ -0,0 +1,187 @@
+			Eventually(controlPlaneClient.Create).
+				WithContext(ctx).
+				WithArguments(kyma).
+				Should(Succeed())
+			By("Then the Kyma CR is in a `Ready` state on control-plane cluster")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+				Should(Succeed())
+			By("And the Kyma CR is in `Ready` sate on runtime cluster")
+			Eventually(CheckRemoteKymaCR).
+				WithContext(ctx).
+				WithArguments(remoteNamespace, []v1beta2.Module{}, runtimeClient, v1beta2.StateReady).
+				Should(Succeed())
+		})
+		It("When a module named `template-operator` is enabled", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", "regular", runtimeClient).
+				Should(Succeed())
+			By("Then the template-operator manager is deployed on runtime cluster")
+			Eventually(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system", "apps/v1",
+					"Deployment", runtimeClient).
+				Should(Succeed())
+			By("And the Kyma CR is in a `Ready` state")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateReady).
+				Should(Succeed())
+		})
+
+		It("When the module `template-operator` is disabled with an existing finalizer", func() {
+			Eventually(SetFinalizer).
+				WithContext(ctx).
+				WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+					[]string{"sample.kyma-project.io/finalizer", "blocking-finalizer"}, runtimeClient).
+				Should(Succeed())
+			Eventually(DisableModule).
+				WithContext(ctx).
+				WithArguments(defaultRemoteKymaName, remoteNamespace, "template-operator", runtimeClient).
+				Should(Succeed())
+			By("Then Kyma CR is in state `Processing`")
+			Eventually(CheckKymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, v1beta2.StateProcessing).
+				Should(Succeed())
+			By("And Manifest CR is in state `Deleting`")
+			Eventually(CheckManifestIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), "template-operator", controlPlaneClient, v2.StateDeleting).
+				Should(Succeed())
+			By("And Sample CR is not removed")
+			Eventually(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("sample-yaml", "kyma-system", "operator.kyma-project.io/v1alpha1", "Sample",
+					runtimeClient).
+				Should(Succeed())
+
+			By("And the template-operator manager Deployment is not removed on runtime cluster")
+			Eventually(CheckIfExists).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system", "apps/v1",
+					"Deployment", runtimeClient).
+				Should(Succeed())
+			By("And it is possible to update the Sample CR in the runtime cluster")
```

Comment:
This check to be honest, is not necessary.

If a resource is under delete, normally the user will not update spec, and the operator also should not design to react to any spec change when resource is under delete.

Reply:
We clarified the description of the original AC is misleading, Scenario 1: Test module CR still possible to be updated during deleting state, it's not mean test module CR update, but test module resources update.